### ISSUE TITLE
Tagger aborts when file selected

### ISSUE THREAD

**Issue by ADBeveridge** [nlogozzo/NickvisionTagger#23]
I opened Tagger and it loaded 4 files. I selected, say B.mp3 (out of A.mp3, B.mp3, C.mp3, and D.mp3) and then went into Files and moved A.mp3 into the trash. Returning to Tagger, I selected the A.mp3 row, and the program faulted. Here is the error message from the cl:
```
terminate called after throwing an instance of 'std::filesystem::__cxx11::filesystem_error'
  what():  filesystem error: cannot get file size: No such file or directory [/home/ad/Music/Dog Barking.mp3]
```

**Comment by nlogozzo** [comment]
Well it failed since the file was deleted. 
Tagger doesn't watch the filesystem for changes as most file operations regarding a tag occur within the program. 
That is why the refresh button is provided, for users who willingly modify file locations outside the program, to rescan the directory for physical changes.

**Comment by ADBeveridge** [comment]
I looked at a experimental [file system watcher](https://github.com/sol-prog/cpp17-filewatcher), that seems to be pretty good, or you could use inotify (type `man inotify`). If this is also being developed for Mac, you could add also a kqueue backend, in addition to the inotify or cpp17-filewatcher.

**Comment by nlogozzo** [comment]
https://github.com/nlogozzo/NickvisionTagger/commit/4bcafccdf50d4820b6db279c178ac7e4ae84c37e

**Comment by ADBeveridge** [comment]
Or use gvfs (part of gtk)

**Comment by ADBeveridge** [comment]
Oh, just saw commit ...never mind

**Comment by nlogozzo** [comment]
Yes, the next update for Tagger will see a rewrite in Qt to allow for cross-platform capabilities.

**Comment by nlogozzo** [comment]
![image](https://user-images.githubusercontent.com/17648453/182765824-e7fb5f63-459c-4b2a-98eb-fc178641bbeb.png)

**Comment by ADBeveridge** [comment]
I see...I thought this was GTK+. Qt is on my next framework to learn :)

**Comment by ADBeveridge** [comment]
Since this has been fixed, this can be close right?

**Comment by nlogozzo** [comment]
It was Gtk. However it is in the process of being rewritten in Qt for the next update. Which is what you see in the screenshot above ^. The development can be seen in the dev branch.

**Comment by nlogozzo** [comment]
There is no need to close the issue yet. Once the dev branch is merged into the master branch, signifying the new release, the issue will close automatically since I have linked it to the commit.

**Comment by nlogozzo** [comment]
https://github.com/nlogozzo/NickvisionTagger/releases/tag/2022.8.0
